core/src/main/scala/org/bykn/bosatsu/codegen/python/PythonGen.scala

```diff
@@ -569,7 +569,7 @@ object PythonGen {
     // if we have a top level let rec with the same name, handle it more cleanly
     val nmVeEnv =
       me match {
-        case Let(Right((n1, r)), inner, Local(n2)) if ((n1 === name) && (n2 === name)) =>
+        case Let(Right((n1, RecursionKind.NonRecursive)), inner, Local(n2)) if ((n1 === name) && (n2 === name)) =>
```

Comment:
this is a change that seems like it matters... but there is a fallback in any case.